### PR TITLE
Support SIGHUP configuration reloading

### PR DESCRIPTION
**Pull request by portertech**
**Description:**

Added SIGHUP UNIX signal handling to reload configuration (internal collector restart). I reused the configuration source notification reloading logic, a very small change.

**Link to tracking Issue:**

This is a possible solution/implementation for https://github.com/open-telemetry/opentelemetry-collector/issues/5966

**Testing:**

Added a basic SIGHUP signal test to ensure the collector finds itself in a running state.

Manual testing demonstrates that the SIGHUP signal successfully triggers the existing configuration reloading logic.

``` bash
$ ./bin/otelcorecol_* --config ./examples/local/otel-config.yaml
```

``` bash
$ ps aux | grep otel
portert+  194518  0.3  0.2 1263472 36692 pts/3   SNl+ 12:55   0:00 ./bin/otelcorecol_linux_amd64 --config ./examples/local/otel-config.yaml
portert+  194534  0.0  0.0  18676   648 pts/4    SN+  12:55   0:00 grep --color=auto otel
$ kill -HUP 194518
```

From: https://gist.github.com/portertech/63a33aca67bc0468120abc924eda49e3

```
2022-08-30T12:01:39.918-0700    info    service/service.go:129  Everything is ready. Begin running and processing data.
2022-08-30T12:04:42.226-0700    info    service/collector.go:201        Received signal from OS {"signal": "hangup"}
2022-08-30T12:04:42.226-0700    warn    service/collector.go:158        Config updated, restart service
2022-08-30T12:04:42.226-0700    info    service/service.go:138  Starting shutdown...
``` 

**Documentation:**

Will need to update documentation to cover the newly supported UNIX signal (SIGHUP). This functionality is very common.

### CODE REVIEW

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
Would like to wait for the issue to propose a solution.

File: service/collector.go [hunk 1]
```diff
@@ -176,21 +194,22 @@ LOOP:
...
 			col.service.telemetrySettings.Logger.Info("Received signal from OS", zap.String("signal", s.String()))
-			break LOOP
+			switch s {
+			case syscall.SIGHUP:
+				if err := col.reloadConfiguration(ctx); err != nil {
+					return err
+				}
+			default:
+				break LOOP
+			}
```
Comment (bogdandrutu) [inline service/collector.go:212]: This is a simple `if` correct?

File: service/collector.go [hunk 2]
```diff
@@ -154,6 +154,21 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	return nil
 }
 
+func (col *Collector) reloadConfiguration(ctx context.Context) error {
```
Comment (bogdandrutu) [inline service/collector.go:157]: I think we should make this no-op if the retrieved config is same as current config. Maybe we can add that later, but would be good to have it.

**Comment by bogdandrutu** [comment]
@swiatekm-sumo any way you can talk to the author, or pickup this?

**Comment by swiatekm-sumo** [comment]
> @swiatekm-sumo any way you can talk to the author, or pickup this?

@portertech is on vacation until Monday. I can pick it up in the meantime if we're in a big hurry though.

File: service/collector.go (hunk 2 above)
Comment (portertech) [reply service/collector.go:157]: I'm in favour of making this enhancement at a later time.

**Comment by portertech** [comment]
@bogdandrutu any other changes required?

**Review by bogdandrutu — APPROVED** [review]
With the small nit left.

**Pull request merged**